### ISSUE TITLE
Invalid Configuration, new React Native version

### ISSUE THREAD

**Issue by rochapablo** [OneSignal/react-native-onesignal#805]
**Description:**

New React Native version gave me the fallow warning:

```
warn Package react-native-onesignal has been ignored because it contains invalid configuration. Reason: Unknown option dependency.platforms.ios.sourceDir with value ""./ios"" was found. This is either a typing error or a user mistake. Fixing it will remove this message.
```

**Environment**

```
"react": "16.8.6",
"react-native": "0.60.3",
"react-native-onesignal": "^3.2.14",
```

**Steps to Reproduce Issue:**

1. Install the OneSignal SDK using npm into your project
2. Build the project
3. Check for the logs

**Comment by blueromans** [comment]
Same issue on 0.60.4

**Comment by TolgaCagin** [comment]
@blueromans  temporary solution is manual linking, 0.60.4

**Comment by blueromans** [comment]
@TolgaCagin Can you describe your way? Because i have try manual linking one more time.

**Comment by TolgaCagin** [comment]
@blueromans , just as usual, dont use react-native link & dont add to react-native-config.js , just manual add required line to settings , app/build.gradle MainApplication.java

**Comment by rochapablo** [comment]
> @blueromans , just as usual, dont use react-native link & dont add to react-native-config.js , just manual add required line to settings , app/build.gradle MainApplication.java

That doesn't work either.

**Comment by yogeshmoradiya121** [comment]
not working manual linking also.
please any one have solution?
this is serious issue
help wanted

**Comment by rgomezp** [comment]
Howdy y'all,
We will be releasing the newest version of the react-native SDK with fixes very soon. If you would like to test the changes now, simply change your react-native-onesignal in your package.json to point to this repository.

```
"react-native-onesignal":"geektimecoil/react-native-onesignal"
```

Would love to hear your feedback prior to the release. Cheers!

EDIT: **PLEASE IGNORE THIS AS THE PLUGIN HAS NOW BEEN PUSHED TO NPM**

**Comment by yogeshmoradiya121** [comment]
after upadting library i faced issue at build time
Could not find androidx.cardview:cardview:28.0.0.
how to solve this??

**Comment by rochapablo** [comment]
> Howdy y'all,
> We will be releasing the newest version of the react-native SDK with fixes very soon. If you would like to test the changes now, simply change your react-native-onesignal in your package.json to point to this repository.
> 
> ```
> "react-native-onesignal":"geektimecoil/react-native-onesignal"
> ```
> 
> Would love to hear your feedback prior to the release. Cheers!

I'll see and get back to you, just a moment...

**[UPDATE]**

It didn't work, follow the log

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:preDebugBuild'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not find androidx.cardview:cardview:28.0.0.
     Required by:
         project :app > project :react-native-onesignal
   > Could not find androidx.cardview:cardview:28.0.0.
     Required by:
         project :app > project :react-native-onesignal > com.onesignal:OneSignal:3.11.1
```

**Comment by thedivac** [comment]
I am having this issue with react-native 0.60.4 and react-native-onesignal 3.3.0:

` Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_RCTOneSignalExtensionService", referenced from:
      objc-class-ref in NotificationService.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

**Comment by simonbuerger** [comment]
I worked around this issue with 0.60 and jetifying for now by doing this in app/build.gradle. Not sure if the new IAM stuff will work, but everything else seems ok.

```
    implementation (project(':react-native-onesignal')) {
        exclude group: 'com.android.support', module: 'cardview-v7'
    }

    implementation "androidx.cardview:cardview:1.0.0"
```

**Comment by rochapablo** [comment]
> I worked around this issue with 0.60 and jetifying for now by doing this in app/build.gradle. Not sure if the new IAM stuff will work, but everything else seems ok.
> 
> ```
>     implementation (project(':react-native-onesignal')) {
>         exclude group: 'com.android.support', module: 'cardview-v7'
>     }
> 
>     implementation "androidx.cardview:cardview:1.0.0"
> ```

Thanks! Doing that, I'm able to build ```("react-native-onesignal": "^3.3.0")```.

Well, some observations...

On Android Studio, even having build with success I got some error (which I lost the log) but I had to do the following steps https://stackoverflow.com/a/49645970/1958882.

And, running through react native command `react-native run-android`, even having build success I also got

```
error React Native CLI uses autolinking for native dependencies, but the following modules are linked manually:
  - react-native-onesignal (to unlink run: "react-native unlink react-native-onesignal")
This is likely happening when upgrading React Native from below 0.60 to 0.60 or above. Going forward, you can unlink this dependency via "react-native unlink <dependency>" and it will be included in your app automatically. If a library isn't compatible with autolinking, disregard this message and notify the library maintainers.
```

Now I'm going to see if the push notification it's working...

**[UPDATE]**

Yep, push notification working!

**Comment by rgomezp** [comment]
Howdy y'all,
Jetifier should be handling this automatically so it must not be working as intended. Please double check that you have followed the [AndroidX migration instructions](https://developer.android.com/jetpack/androidx/migrate) correctly

**Comment by rochapablo** [comment]
> Howdy y'all,
> Jetifier should be handling this automatically so it must not be working as intended. Please double check that you have followed the [AndroidX migration instructions](https://developer.android.com/jetpack/androidx/migrate) correctly

Just to be clear,

I'm not migrating, but I'm starting a new project with the new react-native version, so AndroidX it's already setup.

```
android.useAndroidX=true
android.enableJetifier=true
``` 

Also, I have `"postinstall": "npx jetify"` always running.

**Comment by EgidioCaprino** [comment]
Anyone still having this issue? We've tried @simonbuerger's solution but it didn't work.

**Comment by jkasten2** [comment]
@rochapablo I am not able to reproduce the "could not find androidx.cardview:cardview:28.0.0." error you are seeing. Can you provide a link to an example app and / or exact steps on reproducing it in a new issue?

Closing this issue as the original "warn Package react-native-onesignal has been ignored because it contains invalid configuration" is now fixed in the [3.3.0 version](https://github.com/geektimecoil/react-native-onesignal/releases)

If you are still see any issues after updating to 3.3.1 please create a new issue

**Comment by yogeshmoradiya121** [comment]
@josh kasten
Build is generated but can user register to one signal

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
